src/main/java/duke/tasks/Deadline.java

```diff
@@ -0,0 +1,82 @@
+package duke.tasks;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+import duke.dukeExceptions.DukeException;
+
+/**
+ * Represents a Deadline in Duke.
+ * 
+ * @author Ramanathan Kumarappan
+ */
+public class Deadline extends Task {
+    private LocalDate date;
+    private LocalTime time;
+    private boolean hasTime;
+
+    /**
+     * Constructor for Deadline.
+     * 
+     * @param description - The description of the Deadline.
+     * @param by - The date and time for the deadline (in YYYY-MM-DD HHMM format).
+     * @throws DukeException - When the description or by date is invalid.
+     */
+    public Deadline(String description, String by) throws DukeException {
```

Comment:
Now, this constructor has two responsibilities: parsing the `by` string and creating the `Deadline`. Would defining another method for the parse, and making the constructor a `LocalDate` as a parameter make things clearer?